components/UI/Datalist.js

```diff
@@ -0,0 +1,114 @@
+import React, { useState, useRef } from 'react';
+
+import { ServiceTable } from './ServiceTable';
+
+/**
+ * An input component that functions as a datalist and can be controlled with arrow keys and mouse clicks.
+ *
+ * @component
+ * @param {string[]} options - An array of Objects for the datalist.
```

Comment:
If this is an "array of objects", then it should be documented as `@param {Object[]} options`.

~~But instead I'd like to suggest a simplification of this `Datalist` component:
Instead of passing an unnecessary complex array of objects just because this was the use case you had on hand when designing this, let's think of this from the `Datalist` point of view. We, as `Datalist`, don't really know what's in that array of objects; this is the reason why you ended up with this `keysToDisplay` property (and the inferred logic).~~

~~Let's think of it the other way around and look at what `Datalist` actually needs:~~
1. ~~A string to display in the list~~
2. ~~Some sort of identifier, ID, index, key to report back to the consumer component after an entry was selected~~

~~Coming from that, this should actually only need what you said in the very beginning in the documentation: an **array of strings**. Something like:~~
<img width="547" alt="image" src="https://github.com/medienhaus/medienhaus-spaces/assets/706419/f11c892b-935f-467e-9242-be0c0efa1995">

~~And once an entry gets selected, you just report back the key. So when `@andi:medienhaus.dev` gets selected, you call `onSelect(1)`. Since the consumer component passed this `options` array, it obviously _knows_ that `1` is `@andi` in this case.~~

~~And -~~

After writing all of the above I realized, that we want to show more than just a string. It's actually a table with different columns... and yes; in that case it makes sense to do what you're doing here.

At the same time I'm unhappy with the user experience. When typing in user names the table "shakes" a lot. Readability is hard when the widths of columns in a table start shifting around with every key you type.

For example in this case, I think the text in the right column could be aligned to the right. To ensure fixed focus points for the eye.

Can you maybe work on polishing this once again with @andirueckel? I'll leave some more UX-related stuff in the final review comment.